src/transaction/locator_sr.c

```diff
@@ -13812,7 +13812,7 @@ locator_multi_insert_force (THREAD_ENTRY * thread_p, HFID * hfid, OID * class_oi
 
 	  // Add this record to the recdes array and increase the accumulated size.
 	  recdes_array.push_back (local_record);
-	  accumulated_records_size += local_record.length;
+	  accumulated_records_size += DB_ALIGN (local_record.length, DOUBLE_ALIGNMENT);
```

Comment:
```suggestion
	  accumulated_records_size += DB_ALIGN (local_record.length, HEAP_MAX_ALIGNMENT);
	  accumulated_records_size += record_overhead;
```

we need to consider slot overhead for a record:
record_overhead = spage_slot_size ();

and also line 13752 would be
if ((DB_ALIGN (local_record.length, HEAP_MAX_ALIGNMENT) + record_overhead + accumulated_records_size) >= heap_max_page_size)